### PR TITLE
Boltz plugin draft

### PR DESCRIPTION
**Pull request by fiksn**

### CODE REVIEW

File: plugins/main.go [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+	PluginFlags []cli.Flag
+)
 
-func init() {
-	Plugins = map[string]agent_entities.Plugin{
-		"boltz": &boltz.Plugin{},
+func InitPlugins(lnAPI agent_entities.NewAPICall, cmdCtx *cli.Context) error {
+	Plugins = make(map[string]agent_entities.Plugin)
+
+	plugin := boltz.NewBoltzPlugin(lnAPI, cmdCtx)
```
Comment (dcrystalj) [inline plugins/main.go:21]: its more standard to return struct && error.

**Review by dcrystalj — APPROVED** [review]

File: plugins/main.go [hunk 2]
```diff
@@ -4,30 +4,42 @@ import (
 	"fmt"
 
 	agent_entities "github.com/bolt-observer/agent/entities"
-	"github.com/bolt-observer/agent/plugins/boltz"
 	"github.com/urfave/cli"
 )
 
+// Note plugins depend on us, not vice versa (we have no idea about them)
```
Comment (dcrystalj) [inline plugins/main.go:10]: on us ?   who is "them" ?

File: plugins/boltz/main.go [hunk 3]
```diff
@@ -3,15 +3,63 @@ package boltz
...
+		params = chaincfg.SimNetParams
+	}
+
+	resp := &Plugin{
+		ChainParams: &params,
+		BoltzAPI: &boltz.Boltz{
+			URL: cmdCtx.String("boltzurl"),
+		},
+	}
+	if lnAPI == nil {
```
Comment (dcrystalj) [inline plugins/boltz/main.go:98]: Do we really need this check?

File: plugins/boltz/utils.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package boltz
```
Comment (dcrystalj) [inline plugins/boltz/utils.go:1]: can we find better name than utils ? maybe `connection.go`

File: plugins/boltz/main.go [hunk 5]
```diff
@@ -1,17 +1,112 @@
...
+		entropy []byte
+		err     error
+	)
+
+	mnemonic := cmdCtx.String("setmnemonic")
+	// TODO: save this to db
+	if mnemonic != "" {
+		entropy, err = bip39.MnemonicToByteArray(mnemonic)
+		if err != nil {
+			glog.Fatalf("Invalid mnemonic: %v", err)
```
Comment (dcrystalj) [inline plugins/boltz/main.go:82]: Why do we have to fatal here. I suggest we propagate (return) errors and stop at the top if error is present.

File: plugins/boltz/main.go (hunk 3 above)
Comment (fiksn) [reply plugins/boltz/main.go:100]: Refactored everything so err is reported there (yeah know I still need to remove this nil check)

File: entities/utils.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
+package entities
```
Comment (dcrystalj) [inline entities/utils.go:1]: currently only type definitions are part of package entities.

**Review by dcrystalj — APPROVED** [review]

**Pull request merged**